packages/cli/src/logging/logger.service.ts

```diff
@@ -0,0 +1,148 @@
+import { GlobalConfig } from '@n8n/config';
+import callsites from 'callsites';
+import { InstanceSettings } from 'n8n-core';
+import { LoggerProxy, LOG_LEVELS } from 'n8n-workflow';
+import path, { basename } from 'node:path';
+import { inspect } from 'node:util';
+import { Service } from 'typedi';
+import winston from 'winston';
+
+import { isObjectLiteral } from '@/utils';
+
+import { noOp } from './constants';
+import type { LogLocationMetadata, LogLevel, LogMetadata } from './types';
+
+@Service()
+export class Logger {
+	private readonly internalLogger: winston.Logger;
+
+	private readonly level: LogLevel;
+
+	constructor(
+		private readonly globalConfig: GlobalConfig,
+		private readonly instanceSettings: InstanceSettings,
+	) {
+		this.level = this.globalConfig.logging.level;
+
+		const isSilent = this.level === 'silent';
+
+		this.internalLogger = winston.createLogger({
+			level: this.level,
+			silent: isSilent,
+		});
+
+		if (!isSilent) {
+			this.setLevel();
+
+			if (this.outputs.includes('console')) this.setConsoleTransport();
+			if (this.outputs.includes('file')) this.setFileTransport();
+		}
+
+		LoggerProxy.init(this);
+	}
+
+	get outputs() {
+		return this.globalConfig.logging.output.split(',').map((o) => o.trim());
+	}
```

Comment:
If we move the string parsing in the config object, we don't need to do this here.

Reply:
[d3c794a](https://github.com/n8n-io/n8n/pull/11031/commits/d3c794a6192ea03937b61a281d5460dc3f6aee8d)